web/scm_helpers.go

```diff
@@ -194,6 +194,10 @@ func isEmpty(ctx context.Context, sc scm.SCM, repos []*pb.Repository) error {
 // creating a course, approving, changing status of, or deleting
 // a course enrollment or group
 func contextCanceled(ctx context.Context) bool {
+	// debugging context related errors
```

Comment:
Suggest to add logger to the signature and make this logging more permanent, so that we have a better chance of identifying root cause... Maybe also pass in the method name from where we call this func. Or can this be simplified with a new `errors` package.

Reply:
It most certainly can be done easily with introduction of the new errors package, right now it just seems that the context is never seen as canceled by the server, so this method is probably unnecessary. 
Since this type of errors is extremely rare, see no other option than trying to catch it in production logs.